### ISSUE TITLE
Anaconda Cloud and Travis CI build failure - Package missing in current linux-64 channels

### ISSUE THREAD

**Issue by artPlusPlus** [conda/conda-build#881]
I have two packages:

- elemental-core [(anaconda)](https://anaconda.org/artPlusPlus/elemental-core) [(github)](https://github.com/artPlusPlus/elemental-core)
- elemental-kinds [(anaconda)](https://anaconda.org/artplusplus/elemental-kinds) [(github)](https://github.com/artPlusPlus/elemental-kinds)

elemental-kinds depends on elemental-core.

Building elemental-kinds on both [Travis CI (build)](https://travis-ci.org/artPlusPlus/elemental-kinds) and [Anaconda Cloud (build)](https://anaconda.org/artplusplus/elemental-kinds/builds/6/0) fails with the same message:
> Solving package specifications: .
> Error:  Package missing in current linux-64 channels:
>   - elemental-core

If [this Anaconda Cloud build](https://anaconda.org/artplusplus/elemental-core/builds/7/0) of elemental-core completed and uploaded successfully, why does the [linux-64 channel url](https://conda.anaconda.org/artplusplus/linux-64/) not show any builds?

### Facts
- The `artplusplus` channel is explicitly added to the user conda config in both environments using `conda config -add channels artplusplus`
- The output of `conda info -a` during both builds includes the `artplusplus` channel.
- I have attempted to use `conda build --channel artplusplus ./conda_recipe`. Builds fail with same error.
- I have attempted to use the [`install_channels`](http://docs.anaconda.org/build.html#Install_Channels) section in the `.binstar.yml`. Builds fail with same error.

### Assumptions
- The [linux-64 channel url](https://conda.anaconda.org/artplusplus/linux-64/) specified in the Travis CI output seems to indicate no builds are available.
- I believe [this Anaconda Cloud build](https://anaconda.org/artplusplus/elemental-core/builds/7/0) generated and uploaded a valid linux-64 build of elemental-core [here](https://anaconda.org/artPlusPlus/elemental-core/files).
- I do not believe access tokens are needed; my anaconda.org account is the free personal tier, which makes everything public.

### Consulted Documentation
- [Anaconda Build - Build Configuration](http://docs.anaconda.org/build.html#BuildConfiguration)
- [the meta.yml file](http://conda.pydata.org/docs/building/meta-yaml.html)

### Travis Build Output (`conda info -a`)
```
Current conda install:
             platform : linux-64
        conda version : 4.0.5
  conda-build version : 1.20.0
       python version : 3.5.1.final.0
     requests version : 2.9.1
     root environment : /home/travis/miniconda  (writable)
  default environment : /home/travis/miniconda
     envs directories : /home/travis/miniconda/envs
        package cache : /home/travis/miniconda/pkgs
         channel URLs : https://conda.anaconda.org/artplusplus/linux-64/
                        https://conda.anaconda.org/artplusplus/noarch/
                        https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : /home/travis/.condarc
    is foreign system : False
```

### .travis.yml
```
language: python
python:
  - "3.5"
before_install:
  - sudo apt-get update -qq
  # We do this conditionally because it saves us some downloading if the
  # version is the same.
  - if [[ "$TRAVIS_PYTHON_VERSION" == "2.7" ]]; then
      wget https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh -O miniconda.sh;
    else
      wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
    fi
  - bash miniconda.sh -b -p $HOME/miniconda
  - export PATH="$HOME/miniconda/bin:$PATH"
  - hash -r
  - conda config --set always_yes yes --set changeps1 no
  - conda config --add channels artplusplus
  - conda update -q conda
  - conda install -q conda-build
  - conda info -a
install:
  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION pytest
  - source activate test-environment
  - conda build ./conda_recipe
  - conda install --use-local elemental-kinds
script:
  - py.test
```

### Anaconda Build Output (`conda info -a`)
```
Current conda install:
             platform : linux-64
        conda version : 4.0.5
  conda-build version : 1.20.0
       python version : 2.7.11.final.0
     requests version : 2.9.1
     root environment : /opt/miniconda  (writable)
  default environment : /home/dev/env
     envs directories : /opt/miniconda/envs
        package cache : /opt/miniconda/pkgs
         channel URLs : https://conda.anaconda.org/artplusplus/linux-64/
                        https://conda.anaconda.org/artplusplus/noarch/
                        https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : /home/dev/condarc
    is foreign system : False
```
### .binstar.yml
```
package: elemental-kinds
platform:
  - linux-64
  - win-64
  - osx-64
engine:
  - python=3.5
script:
  - conda config --add channels artplusplus
  - conda build ./conda_recipe
build_targets:
  - conda
```

### meta.yml
```
package:
  name: elemental-kinds
  version: "0.1.1dev2"

source:
  path: ../

requirements:
  build:
    - python
    - setuptools
    - elemental-core
  run:
    - python
    - elemental-core

test:
  requires:
    - pytest
  imports:
    - elemental_kinds

about:
  home: https://github.com/artplusplus/elemental-kinds.
  summary: elemental-kinds is a collection of Attribute Kind classes.
  license: Mozilla Public License 2.0 (MPL 2.0)
  license_file: LICENSE
```

**Comment by artPlusPlus** [comment]
I have finally figured out this issue and believe it was due to my own lack of knowledge.

I did not properly understand how labels worked in Anaconda Build/Cloud. The configurations above implicitly pull from the "main" label. However, my configuration for Anaconda CI labeled new builds as "master".

After updating the Anaconda CI configurations to label builds as "main", dependent packages were able to build successfully in both Travis and Anaconda Cloud.
